contracts/libs/trader-pool/TraderPoolInvest.sol

```diff
@@ -90,6 +100,52 @@ library TraderPoolInvest {
         }
     }
 
+    function investTokens(
+        ITraderPool.PoolParameters storage poolParameters,
+        mapping(address => mapping(uint256 => uint256)) storage investsInBlocks,
+        EnumerableSet.AddressSet storage positions,
+        address holder,
+        uint256[] calldata amounts,
+        address[] calldata tokens
+    ) external returns (uint256 toMintLP) {
+        address baseToken = poolParameters.baseToken;
+
+        TraderPool traderPool = TraderPool(address(this));
+
+        for (uint256 i; i < tokens.length; i++) {
+            require(
+                !traderPool.coreProperties().isBlacklistedToken(tokens[i]),
+                "TP: token in blacklist"
+            );
+            uint256 baseAmount;
+            IERC20(tokens[i]).transferFrom(holder, address(this), amounts[i]);
+
+            if (tokens[i] != baseToken) {
+                (baseAmount, ) = traderPool.priceFeed().getNormalizedPriceOut(
+                    tokens[i],
+                    baseToken,
+                    amounts[i]
+                );
+
+                if (positions.contains(tokens[i])) {
+                    emit ActivePortfolioExchanged(baseToken, tokens[i], baseAmount, amounts[i]);
+                } else {
+                    positions.add(tokens[i]);
+                    emit Exchanged(msg.sender, baseToken, tokens[i], baseAmount, amounts[i]);
+                }
+            } else {
+                baseAmount = amounts[i];
+            }
+
+            toMintLP += baseAmount;
```

Comment:
Yeah, all good, but what if there already are positions? We have to account for that when calculating LP price